cmd/sippy/annotatejobruns.go

```diff
@@ -0,0 +1,203 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/openshift/sippy/pkg/api/componentreadiness"
+	crtype "github.com/openshift/sippy/pkg/apis/api/componentreport"
+	"github.com/openshift/sippy/pkg/apis/cache"
+	bqcachedclient "github.com/openshift/sippy/pkg/bigquery"
+	"github.com/openshift/sippy/pkg/componentreadiness/jobrunannotator"
+	"github.com/openshift/sippy/pkg/dataloader/prowloader/gcs"
+	"github.com/openshift/sippy/pkg/flags/configflags"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+
+	"github.com/openshift/sippy/pkg/flags"
+)
+
+type AnnotateJobRunsFlags struct {
+	BigQueryFlags           *flags.BigQueryFlags
+	PostgresFlags           *flags.PostgresFlags
+	GoogleCloudFlags        *flags.GoogleCloudFlags
+	CacheFlags              *flags.CacheFlags
+	ComponentReadinessFlags *flags.ComponentReadinessFlags
+	ConfigFlags             *configflags.ConfigFlags
+	VariantStr              []string
+	Variants                []crtype.Variant
+	Release                 string
+	Label                   string
+	BuildCluster            string
+	StartTimeStr            string
+	StartTime               time.Time
+	Duration                time.Duration
+	MinimumFailure          int
+	DryRun                  bool
+	FlakeAsFailure          bool
+	TextContains            string
+	TextRegex               string
+	PathGlob                string
+	JobRunIDs               *[]int64
+	Remark                  string
+}
+
+func NewAnnotateJobRunsFlags() *AnnotateJobRunsFlags {
+	return &AnnotateJobRunsFlags{
+		BigQueryFlags:           flags.NewBigQueryFlags(),
+		PostgresFlags:           flags.NewPostgresDatabaseFlags(),
+		GoogleCloudFlags:        flags.NewGoogleCloudFlags(),
+		CacheFlags:              flags.NewCacheFlags(),
+		ComponentReadinessFlags: flags.NewComponentReadinessFlags(),
+		ConfigFlags:             configflags.NewConfigFlags(),
+	}
+}
+
+func (f *AnnotateJobRunsFlags) BindFlags(fs *pflag.FlagSet) {
+	f.BigQueryFlags.BindFlags(fs)
+	f.PostgresFlags.BindFlags(fs)
+	f.GoogleCloudFlags.BindFlags(fs)
+	f.CacheFlags.BindFlags(fs)
+	f.ComponentReadinessFlags.BindFlags(fs)
+	f.ConfigFlags.BindFlags(fs)
+	fs.StringArrayVar(&f.VariantStr, "variant", f.VariantStr, "A variant to include to search for job runs. The format of the variant string is [variant]:[value] (e.g. Platform:metal).")
```

Comment:
In the long term, nice to have category... Taking a URL and converting it to a variant set would be awesome.  But not for today.